File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    arr = [[' ' for _ in range(n)] for _ in range(n)]
    for i in range(n):
        str1 = change_num_to_2num(arr1[i], n)
        str2 = change_num_to_2num(arr2[i], n)
        for j in range(n):
            if str1[j] == '1' or str2[j] == '1':
                arr[i][j] = '#'
    return [''.join(line) for line in arr]

def change_num_to_2num(num, max_len):
    answer = ''
    for i in range(max_len-1, -1, -1):
        if num - 2**i >= 0:
            answer += '1'
            num -= 2**i
        else:
            answer += '0'
    return answer
        
        
# 이진수로 변환
# 11111 일때 2^4+2^3+2^2+2^1+2^0
# n % 2^5

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.js
function solution(n, arr1, arr2) {
    const answer = Array(n).fill().map(el => Array(n).fill(' '));
    for (let i = 0; i < n; i++){
        const str1 = numTo2Num(arr1[i], n);
        const str2 = numTo2Num(arr2[i], n);
        for (let j = 0; j < n; j++){
            if (str1[j] === '1' || str2[j] === '1'){
                answer[i][j] = '#'
            }
        }
    }
    return answer.map(line => line.join(''));
}

const numTo2Num = (num, maxLen) => {
    let answer = '';
    for (let i = maxLen-1; i >= 0; i--){
        if (num - Math.pow(2, i) >= 0){
            answer += '1';
            num -= Math.pow(2, i)
        }else {
            answer += '0'
        }
    }
    return answer
}